ledger/sandbox/src/main/scala/com/digitalasset/platform/store/dao/events/TransactionsReader.scala

```diff
@@ -15,23 +15,18 @@ private[dao] object TransactionsReader {
   def apply(
       dispatcher: DbDispatcher,
       dbType: DbType,
-      executionContext: ExecutionContext): TransactionsReader =
-    new TransactionsReader {
-      override def lookupFlatTransactionById(
-          transactionId: TransactionId,
-          requestingParties: Set[Party],
-      ): Future[Option[GetFlatTransactionResponse]] = {
-        val query = EventsTable
-          .prepareLookupFlatTransactionById(transactionId, requestingParties)
-        dispatcher
-          .executeSql(
-            description = "lookup_flat_transaction_by_id",
-            extraLog = Some(s"tx: $transactionId, parties = ${requestingParties.mkString(", ")}"),
-          ) { implicit connection =>
-            query.as(EventsTable.flatEventParser.*)
-          }
-          .map(RawFlatEvent.aggregate)(executionContext)
-      }
+  )(implicit executionContext: ExecutionContext): TransactionsReader =
+    (transactionId: TransactionId, requestingParties: Set[Party]) => {
+      val query = EventsTable
+        .prepareLookupFlatTransactionById(transactionId, requestingParties)
+      dispatcher
+        .executeSql(
+          description = "lookup_flat_transaction_by_id",
+          extraLog = Some(s"tx: $transactionId, parties = ${requestingParties.mkString(", ")}"),
+        ) { implicit connection =>
+          query.as(EventsTable.flatEventParser.*)
+        }
+        .map(RawFlatEvent.aggregate)
```

Comment:
It was done on purpose because I'm adding another method here, please undo this!

Reply:
Will do!

Reply:
I'll leave this PR for now and pick it up again when #5089 is merged, to ensure I don't cause conflicts.

Reply:
If you just drop this file from the change list I believe there should be no conflicts, but thanks for caring. :slightly_smiling_face:

Reply:
Honestly, I'm not a great fan of using the functional interface functionality, because it makes it harder to find where a certain type is being instantiated. At least IntelliJ doesn't cope well (or at all) with it.

Reply:
I switched it because IntelliJ told me to. Happy to avoid doing this in the future if people find it confusing.

Reply:
Threw away this commit.